implementations/elixir/ockam/ockam_vault_software/native/vault/software/nifs.c

```diff
@@ -482,7 +482,7 @@ static ERL_NIF_TERM ecdh(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
 }
 
 static ERL_NIF_TERM hkdf_sha256(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
-    if (4 != argc) {
+    if (4 != argc && 3 != argc) {
```

Comment:
what is the expected `argc` here?

Reply:
Its both because of how elixir works.

[Here](https://github.com/ockam-network/ockam/pull/622/files#diff-4014ccd33ab221998d1257af7ceac2a01440c8c7d1bc55de2772fd5128f39f6eR86) is the one for 4 inputs

[Here](https://github.com/ockam-network/ockam/pull/622/files#diff-4014ccd33ab221998d1257af7ceac2a01440c8c7d1bc55de2772fd5128f39f6eR90) is the added one for 3 inputs

Reply:
Yes, ikm is optional

Reply:
got it, thanks!